docs/prover/techniques/index.md

```diff
@@ -1,31 +1,121 @@
+   edges that become unreachable through the initial edge removal.
+
+```{figure} split-step.png
+:name: single_split()
+:alt: A single splitting step
+:align: center
+:height: 300px
+
+Illustration of a single splitting step
+```
+
+
+There is an internal heuristic deciding which branching nodes to pick for each
+single splitting step.
+
+The following pseudo-code illustrates how Certora prover applies the single splitting 
+in a recursive fashion.
+
+```{code-block}
+:name: recursive splitting algorithm
+:caption: "Recursive splitting algorithm as pseudo code"
+
+Input: input_program_cfg
+
+worklist = []
+worklist.add([input_program_cfg, 0])
+
+while (worklist != [])
+    [cfg, current_depth] = worklist.pop()
+
+    res = smt_check(cfg, get_timeout_for(current_depth))
+    when (res) 
+        [SAT, model] -> return [SAT, model]
+        UNSAT -> continue
+        TIMEOUT -> 
+            if (current_depth == max_depth)
+                return timeout
+            else
+                [split_1, split_2] = split_single(cfg)
+                worklist.add([split_1, current_depth + 1])
+                worklist.add([split_2, current_depth + 1])
+return UNSAT
+```
+
+Intuitively, the algorithm explores the tree of all possible recursive
+splittings along a fixed sequence of split points up to the maximum splitting
+depth. We call the splits at maximum splitting depth *split leafs*. The
+exploration stops in either of the following three cases:
+ - if one split was found that is SAT (reasoning: if one split is SAT, then the
+  original program must be SAT, since the behavior of the split is replayable in
+  the original program)
+ - if all splits have been shown to be unsat
+ - if solving on a split leaf has timed out (except if 
+   {ref}`-dontStopAtFirstSplitTimeout` has been set)
+
+The settings with which the user can influence this process are the
+following (each links to a more detailed description of the option):
 
-% TODO write this -- tracked in https://certora.atlassian.net/browse/DOC-351
+ - [Maximum split depth](-depth) controls the maximum splitting depth.
+ - [Medium timeout](-mediumTimeout) controls the timeout that is applied when
+   checking splits that are not split leafs, i.e., that are not at the maximum
+   depth. 
+ - [Smt timeout](--smt_timeout) controls the timeout that is used to solve split 
+   leafs; if this is exceeded, the Prover will give up with a TIMEOUT 
+   result, unless [the corresponding setting](-dontStopAtFirstSplitTimeout) says 
+   to go on.
+ - Setting the [initial splitting depth](-smt_initialSplitDepth) to a value 
+   above 0 will make the Prover skip the checking and immediately enumerate all 
+   splits up to that depth.
 
 (storage-and-memory-analysis)=
-# Storage and Memory Analysis
+# Storage and Memory analysis
 
 The Certora Prover works on EVM bytecode as its input. To the bytecode, the
 address space of both Storage and Memory are flat number lines. That two
```

Comment:
I don't think Storage and Memory should be capitalized.

Reply:
The idea of capitalizing was to indicate that I'm referring to the terms as defined by EVM, not to the general concepts (in which case I wouldn't be sure what the difference between the two even is exactly) ... 

Maybe I make them glossary references instead of capitalizing (first occurrence at least .. make new glossary entries, too, if necessary)?.. 
I'll try that for now, open to comments.

Reply:
I'll use the terms "EVM storage" and "EVM memory" instead of the capitalized Storage and Memory ..